client/signup/steps/wp-for-teams-site/index.jsx

```diff
@@ -0,0 +1,340 @@
+					{
+						site: props.step.errors[ 0 ].message,
+					},
+					true
+				);
+			}
+		}
+
+		this.formStateController = new formState.Controller( {
+			fieldNames: [ 'site', 'siteTitle' ],
+			sanitizerFunction: this.sanitize,
+			validatorFunction: this.validate,
+			onNewState: this.setFormState,
+			onError: this.handleFormControllerError,
+			debounceWait: VALIDATION_DELAY_AFTER_FIELD_CHANGES,
+			hideFieldErrorsOnChange: true,
+			initialState: initialState,
+		} );
+
+		this.state = {
+			form: this.formStateController.getInitialState(),
+			submitting: false,
+		};
+	}
+
+	componentWillUnmount() {
+		this.save();
+	}
+
+	sanitizeSubdomain = domain => {
+		if ( ! domain ) {
+			return domain;
+		}
+		return domain.replace( /[^a-zA-Z0-9]/g, '' ).toLowerCase();
+	};
+
+	sanitize = ( fields, onComplete ) => {
+		const sanitizedSubdomain = this.sanitizeSubdomain( fields.site );
+		if ( fields.site !== sanitizedSubdomain ) {
+			onComplete( { site: sanitizedSubdomain } );
+		}
+	};
+
+	validate = ( fields, onComplete ) => {
+		const messages = {};
+
+		if ( isEmpty( fields.siteTitle ) ) {
+			messages.siteTitle = {
+				123: this.props.translate( 'Please enter your team or project name.' ),
+			};
+		}
+
+		wpcom.undocumented().sitesNew(
+			{
+				blog_name: fields.site,
+				blog_title: fields.siteTitle,
+				validate: true,
+			},
+			function( error, response ) {
+				debug( error, response );
+
+				if ( error && error.message ) {
+					if ( fields.site && ! includes( siteUrlsSearched, fields.site ) ) {
+						siteUrlsSearched.push( fields.site );
+
+						analytics.tracks.recordEvent(
+							'calypso_signup_wp_for_teams_site_url_validation_failed',
+							{
+								error: error.error,
+								site_url: fields.site,
+							}
+						);
+					}
+
+					timesValidationFailed++;
+
+					messages.site = {
+						[ error.error ]: error.message,
+					};
+				}
+				onComplete( null, messages );
+			}
+		);
+	};
+
+	setFormState = state => {
+		this.setState( { form: state } );
+	};
+
+	resetAnalyticsData = () => {
+		siteUrlsSearched = [];
+		timesValidationFailed = 0;
+	};
+
+	handleSubmit = event => {
+		event.preventDefault();
+
+		this.setState( { submitting: true } );
+
+		this.formStateController.handleSubmit(
+			function( hasErrors ) {
+				const site = formState.getFieldValue( this.state.form, 'site' );
+				const siteTitle = formState.getFieldValue( this.state.form, 'siteTitle' );
+
+				this.setState( { submitting: false } );
+
+				if ( hasErrors ) {
+					return;
+				}
+
+				analytics.tracks.recordEvent( 'calypso_signup_wp_for_teams_site_step_submit', {
+					unique_site_urls_searched: siteUrlsSearched.length,
+					times_validation_failed: timesValidationFailed,
+				} );
+
+				this.resetAnalyticsData();
+
+				this.props.submitSignupStep( {
+					stepName: this.props.stepName,
+					form: this.state.form,
+					site,
+					siteTitle,
+				} );
+
+				this.props.goToNextStep();
+			}.bind( this )
```

Comment:
Just for consistency with the rest of your component, it might make sense to use a fat arrow function here rather than explicitly binding `this`.